### PR TITLE
Search notes by id

### PR DESCRIPTION
**Pull request by martindsq**
#173 

Implemented search notes by id just like we do for authors

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -8,6 +8,8 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ## [Unreleased]
 
+- Discover screen can now search notes by id.
+- Localized relative times on note cards.
```
Comment (mplorentz) [inline CHANGELOG.md:12]: Looks like an extra line snuck in here. It is already listed below.
```suggestion
```

**Review by mplorentz — APPROVED** [review]
Ah, I'm so glad we finally have this. To me it's confusing to have a function `note(fromPublicKey:)` and an initializer on the PublicKey model because the note ID isn't a public key, it's really a unique ID generated based on the contents. I would like to see a `NoteIdentifier` type that could be initialized via a bech32 `note123...` style ID, and we could add initializers for `nevent` and `hex` IDs in the future.

But functionally this is a huge improvement and I don't want to delay that. Maybe we can add a note about this refactoring to the ticket or create a new one. I would also like to have a ticket for nevent handling although maybe that should go into the new idea form in Notion.

**Comment by mplorentz** [comment]
Also those unit tests are failing on main too. I made a note to myself to disable them tomorrow and create a ticket to get them fixed.

**Comment by martindsq** [comment]
I created #757 for that, and created #758 as well

**Pull request merged**